appengine/pond_storage/get.py

```diff
@@ -19,18 +19,100 @@
 
 import cgi
 import json
+import logging
 from google.appengine.ext import ndb
+from google.appengine.api import users
 from pond_storage import *
 
+
+def format_top_ducks(top_ducks_info):
+  """Adds an isOwner property to top ducks that are owned by the current user.
+  Args:
+    top_ducks_info: A list of ducks with the best rankings.
+  Returns:
+    A list of dictionaries containing duck information.
+    Example:
+    {
+      'name': 'bob',
+      'duck_key': 'urlsafe duck key',
+      'code': {'js': 'some code', 'opt_xml': 'some xml'},
+      'published': 'true',
+      'isOwner':'True'
+    }
+  """
+  formatted_ducks = []
+  user_key = get_user_key(users.get_current_user())
+  user_ducks = Duck.query(ancestor=user_key).fetch()
+
+  for duck_info in top_ducks_info:
+    matches = [x for x in user_ducks if x.key.urlsafe() == duck_info['duck_key']]
+    if len(matches) > 0:
+      duck_info['isOwner'] = True
+    formatted_ducks.append(duck_info)
+  return formatted_ducks
+
+def get_top_entries(user_entry, count):
+  """Gets the entries with the best rankings.
+  Args:
+    user_entry: The LeaderboardEntry entity requesting top ducks.
+    count: The number of top ducks to get.
+  Returns:
+    A list of LeaderboardEntry entities that are at the top of their leadboard.
+  """
+  entry_query = LeaderboardEntry.query(
+    LeaderboardEntry.leaderboard_key == user_entry.leaderboard_key,
+    LeaderboardEntry.ranking >= 1,
+    LeaderboardEntry.ranking <= count
+  )
+  entry_query.order = ['ranking']
+  return entry_query.fetch()
+
+def get_top_ducks(forms, duck):
+  """Gets a list of the ducks with the best rankings.
+  Args:
+    forms: The field storage object.
```

Comment:
I think this should just take count directly. That way extracting form information is all in the same place.

Reply:
Fixed.